src/mscorlib/src/System/Version.cs

```diff
@@ -145,93 +129,32 @@ public int CompareTo(Object version)
                 throw new ArgumentException(Environment.GetResourceString("Arg_MustBeVersion"));
             }
 
-            if (this._Major != v._Major)
-                if (this._Major > v._Major)
-                    return 1;
-                else
-                    return -1;
-
-            if (this._Minor != v._Minor)
-                if (this._Minor > v._Minor)
-                    return 1;
-                else
-                    return -1;
-
-            if (this._Build != v._Build)
-                if (this._Build > v._Build)
-                    return 1;
-                else
-                    return -1;
-
-            if (this._Revision != v._Revision)
-                if (this._Revision > v._Revision)
-                    return 1;
-                else
-                    return -1;
-
-            return 0;
+            return CompareTo(v);
         }
 
         public int CompareTo(Version value)
         {
-            if (value == null)
-                return 1;
-
-            if (this._Major != value._Major)
-                if (this._Major > value._Major)
-                    return 1;
-                else
-                    return -1;
-
-            if (this._Minor != value._Minor)
-                if (this._Minor > value._Minor)
-                    return 1;
-                else
-                    return -1;
-
-            if (this._Build != value._Build)
-                if (this._Build > value._Build)
-                    return 1;
-                else
-                    return -1;
-
-            if (this._Revision != value._Revision)
-                if (this._Revision > value._Revision)
-                    return 1;
-                else
-                    return -1;
-
-            return 0;
+            return
+                value == null ? 1 :
+                _Major != value._Major ? (_Major > value._Major ? 1 : -1) :
+                _Minor != value._Minor ? (_Minor > value._Minor ? 1 : -1) :
+                _Build != value._Build ? (_Build > value._Build ? 1 : -1) :
+                _Revision != value._Revision ? (_Revision > value._Revision ? 1 : -1) :
+                0;
```

Comment:
I found that this change compiled to less IL and was faster than the previous implementation.

Results of a 100,000,000 iteration microbenchmark (Windows 10 x64):

```
Old: Time: 0.7367762
Old: Time: 0.5709981
Old: Time: 0.5821049
Old: Time: 0.5728211
Old: Time: 0.5516054
New: Time: 0.3191392
New: Time: 0.3199933
New: Time: 0.3205972
New: Time: 0.3213405
New: Time: 0.323658
```